hydra-plutus/src/Hydra/Contract/Util.hs

```diff
@@ -47,3 +48,15 @@ mustNotMintOrBurn TxInfo{txInfoMint} =
     isZero txInfoMint
 {-# INLINEABLE mustNotMintOrBurn #-}
 
+
+
+infix 4 ===!
+
+-- | Checks for exact exuality between two serialized values
+-- Equality on value is very memory intensive as it's defined on associative
+-- lists and Map equality is implemented. Instead we can be more strict and
+-- require EXACTLY the same value and compare using the serialised bytes.
+(===!) :: Value -> Value -> Bool
+(===!) val val' =
```

Comment:
Is `===` or `==!` already taken?

Reply:
No, I first webt with '===' but then thought to add extra sprinkle using '!' because this is a _strict_ check 😀

Reply:
That's fine by me, renamed!

Reply:
Hello!

I was trying to compare results using (===!):
`
(===!) :: LedgerApiV2.Value -> LedgerApiV2.Value -> Bool
(===!) val val' =
  TxBuiltins.serialiseData (LedgerApiV2.toBuiltinData val) == TxBuiltins.serialiseData (LedgerApiV2.toBuiltinData val')
`
But got an error:
`
Eval Error: CodecError (DeserialiseFailure 2663 "BadEncoding (0x0000004231c2aa77,S {currPtr = 0x0000004231c2a04e, usedBits = 3}) \"Forbidden builtin function: (builtin serialiseData)\"")
`
Any idea?

Reply:
You need to run it using protocol version `>= 7.0`. The `serialiseData` builtin was enabled in version 7 of the ledger/plutus interface.

Reply:
Hello @ch1bo ! 
Im still trying to build and run with the serialiseData to compare results of used resources, but I'm getting the same error "Forbidden builtin function: (builtin serialiseData)\".
I just updated Plutus Apps to 1.1.0, the last release and updated cabal and project file according to the files in Plutus Apps.
What else I need to Update?
And also, I let a question for you in cardano stakoverflow... I would be happy if you can help me with that too
Thank you!!!!!

Reply:
It's not "an own plutus", but we are *not* using plutus-apps. I don't know where you need to do the setting there, but you want to use protocol version `>= 7` when you run the scripts by your plutus interpreter (e.g. in a `cardano-ledger` or a simulator or so).

Reply:
Thank you brother! 
I just realize that basic stuff about the protocol version I was using to do the evaluation. Now it works!
The method with serialiseData is the best I tested by now.  With 20 different tokens, using your way, ExMemory 7294670, mine: ExMemory 9089316, and with (==), ExMemory 13382024. As you can see it serialiseData is almost the half of the normal equality. 
Im about to use your way of comparing in a production contracts Im developing. I really need the extra free mem I getting now.
But still I dont really get in full clarity in what kind of scenarios VALUES are not normalized, not ordered and with cs or tn repeated and this way of comparing can fail. 
Im using off chain code to create the transactions with lucid framework in a frontend. 
Where comes the part that creates the list of values, normalized or not? 
I need to understand a little more that to feel more safe in using your way or mine.
Thank you so much